core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -203,6 +203,14 @@ public SqlCodeGenerator(
         }
     }
 
+    public static void coerceRuntimeConstantType(Function func, short type, SqlExecutionContext context, CharSequence message, int pos) throws SqlException {
+        if (ColumnType.isUndefined(func.getType())) {
+            func.assignType(type, context.getBindVariableService());
+        } else if ((!func.isConstant() && !func.isRuntimeConstant()) || !(ColumnType.isAssignableFrom(func.getType(), type))) {
```

Comment:
nit: no need for extra round brackets:
```suggestion
        } else if ((!func.isConstant() && !func.isRuntimeConstant()) || !ColumnType.isAssignableFrom(func.getType(), type)) {
```